File: insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/app_layer/status/GetPumpStatusRegisterMessage.java
package info.nightscout.androidaps.plugins.pump.insight.app_layer.status;

import info.nightscout.androidaps.plugins.pump.insight.app_layer.AppLayerMessage;
import info.nightscout.androidaps.plugins.pump.insight.app_layer.Service;
import info.nightscout.androidaps.plugins.pump.insight.descriptors.MessagePriority;
import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf;

public class GetPumpStatusRegisterMessage extends AppLayerMessage {

    private boolean operatingModeChanged;
    private boolean batteryStatusChanged;
    private boolean cartridgeStatusChanged;
    private boolean totalDailyDoseChanged;
    private boolean activeBasalRateChanged;
    private boolean activeTBRChanged;
    private boolean activeBolusesChanged;

    public GetPumpStatusRegisterMessage() {
        super(MessagePriority.NORMAL, false, false, Service.STATUS);
    }

    @Override
    protected void parse(ByteBuf byteBuf) {
        operatingModeChanged = byteBuf.readBoolean();
        batteryStatusChanged = byteBuf.readBoolean();
        cartridgeStatusChanged = byteBuf.readBoolean();
        totalDailyDoseChanged = byteBuf.readBoolean();
        activeBasalRateChanged = byteBuf.readBoolean();
        activeTBRChanged = byteBuf.readBoolean();
        activeBolusesChanged = byteBuf.readBoolean();
    }

    public boolean isOperatingModeChanged() {
        return this.operatingModeChanged;
    }

    public boolean isBatteryStatusChanged() {
        return this.batteryStatusChanged;
    }

    public boolean isCartridgeStatusChanged() {
        return this.cartridgeStatusChanged;
    }

    public boolean isTotalDailyDoseChanged() {
        return this.totalDailyDoseChanged;
    }

    public boolean isActiveBasalRateChanged() {
        return this.activeBasalRateChanged;
    }

    public boolean isActiveTBRChanged() {
        return this.activeTBRChanged;
    }

    public boolean isActiveBolusesChanged() {
        return this.activeBolusesChanged;
    }
}

File: insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/app_layer/status/GetPumpStatusRegisterMessage.kt
package info.nightscout.androidaps.plugins.pump.insight.app_layer.status

import info.nightscout.androidaps.plugins.pump.insight.app_layer.AppLayerMessage
import info.nightscout.androidaps.plugins.pump.insight.app_layer.Service
import info.nightscout.androidaps.plugins.pump.insight.descriptors.MessagePriority
import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf

class GetPumpStatusRegisterMessage : AppLayerMessage(MessagePriority.NORMAL, false, false, Service.STATUS) {

    internal var isOperatingModeChanged = false
        private set
    internal var isBatteryStatusChanged = false
        private set
    internal var isCartridgeStatusChanged = false
        private set
    internal var isTotalDailyDoseChanged = false
        private set
    internal var isActiveBasalRateChanged = false
        private set
    internal var isActiveTBRChanged = false
        private set
    internal var isActiveBolusesChanged = false
        private set

    override fun parse(byteBuf: ByteBuf) {
        byteBuf.let {
            isOperatingModeChanged = it.readBoolean()
            isBatteryStatusChanged = it.readBoolean()
            isCartridgeStatusChanged = it.readBoolean()
            isTotalDailyDoseChanged = it.readBoolean()
            isActiveBasalRateChanged = it.readBoolean()
            isActiveTBRChanged = it.readBoolean()
            isActiveBolusesChanged = it.readBoolean()
        }
    }
}